util/reggen/register.py

```diff
@@ -178,6 +178,11 @@ def __init__(self,
         self.update_err_alert = update_err_alert
         self.storage_err_alert = storage_err_alert
 
+        # for multreg
+        self.mname = None
+        self.creg_idx = None
+        self.dv_packed_fields = []  # type List[Field]
```

Comment:
This is exactly the thing I'm trying to avoid: a register shouldn't need to know or care about a surrounding multi-register. Please could you try to re-jig this so that the template code that's iterating over registers and multi-registers passes this information down ("by the way, you're item 3 of a multi register called FOO") rather than storing that on the register itself.

Reply:
it's not very convenient to get instance name. I created a function get_inst_to_reg_dict in the tpl to make it easier and also keep the register class simple. PTAL

Reply:
Looks good, thanks!